adal/src/main/java/com/microsoft/aad/adal/AuthenticationRequestState.java

```diff
@@ -31,12 +31,15 @@
     private boolean mCancelled = false;
 
     private AuthenticationRequest mRequest = null;
-    
+
+    private final APIEvent mAPIEvent;
+
     public AuthenticationRequestState(int requestCallbackId, AuthenticationRequest request,
-            AuthenticationCallback<AuthenticationResult> delegate) {
+            AuthenticationCallback<AuthenticationResult> delegate, final APIEvent apiEvent) {
```

Comment:
> final APIEvent apiEvent [](start = 66, length = 24)

I think i understand why you want to add the API event here, but my thought is this is too much for this class. Consider having a map in the AuthenticationContext itself, and use it to keep track of all the events by the request id.